src/services/APIClient.ts

```diff
@@ -16,40 +17,91 @@ export class APIClient {
     });
   }
 
-  async getBlocks(options?: { offset?: number, limit?: number }): Promise<IBlock[]> {
+  async getBlocks(options?: {
+    offset?: number;
+    limit?: number;
+  }): Promise<IBlock[]> {
     const response = await this.axios.get<IBlock[]>('/blocks', {
-      params: options
+      params: options,
     });
 
     return response.data;
   }
 
   async getBlock(blockId: string): Promise<IBlock> {
-    const response = await this.axios.get<{data: IBlock}>(`/blocks/${blockId}`);
+    const response = await this.axios.get<{ data: IBlock }>(
+      `/blocks/${blockId}`
+    );
 
     return response.data.data;
   }
 
   async getLeavesOfBlock(blockId: string): Promise<ILeaf[]> {
-    const response = await this.axios.get<ILeaf[]>(`/blocks/${blockId}/leaves`); 
+    const response = await this.axios.get<ILeaf[]>(`/blocks/${blockId}/leaves`);
 
     return response.data;
   }
 
   async getKeys(): Promise<IKey[]> {
-    const response = await this.axios.get<{data: IKey[]}>('/keys');
+    const response = await this.axios.get<{ data: IKey[] }>('/keys');
     return response.data.data;
   }
 
   async getProofs(keys: string[]): Promise<IProofs | null> {
-    const response = await this.axios.get<{data: IProofs | Record<string, never>}>('/proofs', {
-      params: { keys }
+    const response = await this.axios.get<{
+      data: IProofs | Record<string, never>;
+    }>('/proofs', {
+      params: { keys },
     });
 
-    if(response.data.data.block) {
+    if (response.data.data.block) {
       return response.data.data as IProofs;
     }
 
     return null;
   }
+
+  /**
+   * Uses verifyProofForBlock method of the Chain contract.
+   * @see https://kovan.etherscan.io/address/0x459DF121Ab6Cf1B4C99119cc354d7C79c83Ec8bE#readContract
+   */
+  async verifyProofForBlock<T extends LeafType>(
+    key: string,
+    leafType: T
+  ): Promise<{
+    success: boolean;
+    value: T extends typeof LeafType.TYPE_INTEGER
+      ? number
+      : T extends typeof LeafType.TYPE_FLOAT
+      ? number
+      : string;
```

Comment:
```suggestion
    value: T extends typeof LeafType.TYPE_INTEGER || T extends typeof LeafType.TYPE_FLOAT ? number : string;
```
can that works? looks simpler

Reply:
@DZariusz unfortunately, it doesnt work. Tho we can make it look simpler using function overloading. I'll try that